### ISSUE TITLE
[BUG] Updating a prototype causes changed attributes to save as string instead of their appropriate type

### ISSUE THREAD

**Issue by Tegiminis** [evennia/evennia#3064]
#### Describe the bug
I'm trying to figure out the best way to save representations of custom classes in prototypes.

Here is the prototype in question:

```python
HIVE_BOOMER = {
    "name": "Hive Boomer",
    "accuracy": 1.0,
    "damage": 10,
    "crit": 2.0,
    "mult": 2.0,
    "shots": 1,
    "cooldown": 5,
    "element": "arc",
    "messaging": {
        "attack": "{attacker} lobs a bomb of electrifying energy at {defender}!",
        "hit": "{defender} is shocked by electric current!",
        "crit": "{defender} is knocked back by explosive force!",
    },
    "is_object": False,
}

HIVE_KNIGHT = {
    "key": "hive knight",
    "typeclass": "typeclasses.npc.NPC",
    "weapon": WeaponStats(**HIVE_BOOMER),
    "brain": PatrolBrain,
}
```

This works fine when I spawn the prototype. Brain and weapon are assigned correctly, show as their respective objects. This is consistent with documentation on spawning.

![image](https://user-images.githubusercontent.com/2596300/210649557-b808002a-a2ac-42b8-baf9-89462095472e.png)

However, when I `spawn/update` the prototype, both brain and weapon are converted to strings after their object is constructed. This obviously breaks everything, as they no longer have the correct methods.

![image](https://user-images.githubusercontent.com/2596300/210649456-fbec8ca8-ad0a-47b9-895b-fae655c01de5.png)
---
This occurs regardless of how the class is assigned. I attempted
```py
"args": [("brain", PatrolBrain), ("weapon", WeaponStats(**HIVE_BOOMER))]
```
and
```py
"attrs": [("brain", PatrolBrain), ("weapon", WeaponStats(**HIVE_BOOMER))]
```
And they resulted in identical behavior.

#### To Reproduce
1. Create a prototype with an attribute that contains a class to instantiate on spawn
2. Spawn the prototype
3. Spawn/update the prototype

#### Expected behavior
Spawn/update replaces old object instances with new ones, not string representations of new instances.

#### Environment, Evennia version, OS etc
Evennia 1.0-dev (rev 31e14f2599)
OS: nt
Python: 3.10.4
Twisted: 22.10.0
Django: 4.1.4

#### Notes
Any prototype which uses instances classes will always count as an "older prototype" for purposes of updating.

**Comment by Tegiminis** [comment]
I did some further experimentation on this.

Any attribute that is changed as a result of `spawn/update` will be a string. Not just instanced classes; any value at all, including literals.

Easy repro:

1. Create a prototype of any object and assign it a `"foo": 10` attribute.
2. Spawn the prototype
3. Change the prototype definition to include `"foo": 1` attribute
4. Spawn/update the prototype
5. Enter python mode
6. Check the type of the db attribute with `type(prototype.db.foo)`

**Comment by Tegiminis** [comment]
Made a pull request. Turns out it's an easy fix. Did an initial validation against my use case and it worked correctly (respected types, including custom classes).

![image](https://user-images.githubusercontent.com/2596300/210678646-8ac73be9-1487-48e0-a068-b2b9f9c22c59.png)
![image](https://user-images.githubusercontent.com/2596300/210678678-169c6d5b-6607-4e4e-aaf8-00e7102670e7.png)

**Comment by Griatch** [comment]
Closed in #3066
